File: solutions/day-45-baseball-game/Solution.java
import java.util.*;

class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> stack = new Stack<>();

        for (String op : operations) {
            if (op.equals("+")) {
                stack.push(stack.peek() + stack.get(stack.size() - 2));
            } else if (op.equals("D")) {
                stack.push(2 * stack.peek());
            } else if (op.equals("C")) {
                stack.pop();
            } else {
                stack.push(Integer.parseInt(op));
            }
        }

        int sum = 0;
        for (int score : stack) sum += score;
        return sum;
    }
}

File: solutions/day-45-baseball-game/Solution.cpp
#include <vector>
#include <string>
#include <stack>

class Solution {
public:
    int calPoints(std::vector<std::string>& operations) {
        std::stack<int> st;

        for (const auto& op : operations) {
            if (op == "+") {
                int top = st.top();
                st.pop();
                int newTop = top + st.top();
                st.push(top);
                st.push(newTop);
            } else if (op == "D") {
                st.push(2 * st.top());
            } else if (op == "C") {
                st.pop();
            } else {
                st.push(std::stoi(op));
            }
        }

        int sum = 0;
        while (!st.empty()) {
            sum += st.top();
            st.pop();
        }
        return sum;
    }
};
